ZenPacks/zenoss/LinuxMonitor/HardDisk.py

```diff
@@ -80,15 +85,40 @@ def impacted_object(self):
             return lv
 
     def storage_disk_lun(self):
-        # return the UCS storage disk/virtual drive
-        # if disk_ids contains the IDs of disk/virtual drive
+        """Return the storage disk/virtual drive based on disk_ids."""
+        keywords = set()
         if self.disk_ids:
-            results = ICatalogTool(self.getDmdRoot('Devices')).search(
-                query=Or(*[Eq('searchKeywords', id)
-                           for id in self.disk_ids]))
-
-            for brain in results:
-                try:
-                    yield brain.getObject()
-                except Exception:
-                    continue
+            for disk_id in self.disk_ids:
+                # Add disk_id to keywords to save compatibility with UCS storage
+                keywords.add(disk_id)
+
+                ldisk_id = disk_id.lower()
+                if ldisk_id.startswith('wwn-0x'):
+                    keywords.add(
+                        'has-target-wwn:{}'.format(ldisk_id.strip('wwn-0x'))
+                    )
+        return keyword_search(self.getDmdRoot('Devices'), keywords)
+
+
+class HardDiskIndexableWrapper(ComponentWrapper):
+    implements(IIndexableWrapper)
+    adapts(HardDisk)
+
+    def searchKeywordsForChildren(self):
+        """Return tuple of search keywords for HardDisk objects."""
+        keywords = set()
+
+        disk_ids = self._context.disk_ids
+
+        if disk_ids:
+            for disk_id in self.disk_ids:
+                ldisk_id = disk_id.lower()
+                if ldisk_id.startswith('wwn-0x'):
+                    keywords.add(
+                        'uses-target-wwn:{}'.format(ldisk_id.strip('wwn-0x'))
```

Comment:
Please comment the code and explain why this label is required. it's not obvious.

Reply:
Why not use lstrip() instead of strip() ?